packages/blocks/src/page-block/edgeless/components/toolbar/brush/brush-tool-button.ts

```diff
@@ -88,9 +112,36 @@ export class EdgelessBrushToolButton extends LitElement {
     }
   }
 
+  override connectedCallback() {
+    super.connectedCallback();
+    const observer = new MutationObserver(() => {
+      // add mouse hover event to pen icon
+      this._penIcon.addEventListener(
+        'mouseenter',
+        this._handleMouseEnter.bind(this)
+      );
+      this._penIcon.addEventListener(
+        'mouseleave',
+        this._handleMouseLeave.bind(this)
+      );
+      observer.disconnect();
+    });
+
+    if (!this.shadowRoot) return;
+    observer.observe(this.shadowRoot, { childList: true });
+  }
+
   override disconnectedCallback() {
     this._brushMenu?.dispose();
     this._brushMenu = null;
+    this._penIcon.removeEventListener(
+      'mouseenter',
+      this._handleMouseEnter.bind(this)
+    );
+    this._penIcon.removeEventListener(
+      'mouseleave',
+      this._handleMouseLeave.bind(this)
+    );
```

Comment:
You use `DisposableGroup` when `addEventListener` or use [WithDisposable](https://github.com/toeverything/blocksuite/blob/94f539f59698122cc7f21e1e4db5ebfbe264074f/packages/blocks/src/__internal__/utils/disposable.ts#L41) directly

```ts
  class DisposableClass {
    private _disposables = new DisposableGroup();

    override connectedCallback() {
      super.connectedCallback();
      if (this._disposables.disposed) {
        this._disposables = new DisposableGroup();
      }
    }

    override disconnectedCallback() {
      super.disconnectedCallback();
      this._disposables.dispose();
    }
  }
```